Lib/test/test_posixpath.py

```diff
@@ -242,42 +243,57 @@ def fake_lstat(path):
     def test_expanduser(self):
         self.assertEqual(posixpath.expanduser("foo"), "foo")
         self.assertEqual(posixpath.expanduser(b"foo"), b"foo")
+
+    def test_expanduser_home_envvar(self):
```

Comment:
The following tests test only few corner cases. It would be nice to add a test for a more general case, for example for `env['HOME'] = '/home/eric'` and `env['HOME'] = '/home/eric/'`.